app/models/compound_aliquot.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+# Factory class for creating Aliquots with compound samples in them.
+# At time of writing, called from Request::SampleCompoundAliquotTransfer, in the context of a Request.
+#
+# The inputs are:
+#   request           A Request for transferring source_aliquots into a target receptacle.
+#                     At time of writing, only used for a SequencingRequest, from a multiplex tube into a lane.
+#   source_aliquots   A list of Aliquots that should be transferred into a single Aliquot in a target Receptacle,
+#                     that containes  contain a single compound sample.
+#                     The list of component samples is derived from source_aliquots.
+#                     Some attributes are transferred from the source aliquots onto the compound aliquot.
+#
+class CompoundAliquot
+  include ActiveModel::Model
+
+  DUPLICATE_TAG_DEPTH_ERROR_MSG = "Cannot create compound sample from following samples due to duplicate 'tag depth'"
+  MULTIPLE_STUDIES_ERROR_MSG =
+    'Cannot create compound sample due to the component samples being under different studies.'
+
+  attr_accessor :request, :source_aliquots
+
+  attr_reader :component_samples, :compound_sample
+
+  validate :tag_depth_is_unique
+
+  def initialize(attributes)
+    super
+
+    @component_samples = source_aliquots.map(&:sample)
+  end
+
+  # Check that the component samples in the compound sample will be able to be distinguished -
+  # this is represented by them all having a unique 'tag_depth'
+  def tag_depth_is_unique
+    return unless source_aliquots.pluck(:tag_depth).uniq!
+
+    errors.add("#{DUPLICATE_TAG_DEPTH_ERROR_MSG}: #{component_samples.map(&:name)}")
+  end
+
+  def default_library_type
+    source_aliquots.first.library_type
```

Comment:
I thought we were only setting library type if it was the same for all aliquots. (Sorry about not catching this one sooner, I think I got it mixed up with library_id)